app/Http/Controllers/Admin/PostController.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Repositories\Post\PostRepository;
+use App\Http\Controllers\Controller;
+use Carbon\Carbon;
+use App\Http\Requests\StorePostRequest;
+
+class PostController extends Controller
+{
+    protected $postRepo;
+
+    public function __construct(PostRepository $postRepo)
+    {
+        $this->postRepo = $postRepo;
+    }
+
+    public function index()
+    {
+        $posts = $this->postRepo->getAll();
+        return view('index', ['posts' => $posts]);
+    }
+
+    public function show_posts()
+    {
+        $posts = $this->postRepo->getAllOrderByDESC();
+        return view('admin.posts.posts', ['posts' => $posts]);
+    }
+
+    public function add_post_page()
+    {
+        return view('admin.posts.add_post');
+    }
+
+    public function add_post(StorePostRequest $request)
+    {
+        // Save file on public/images
+        $imageName = Carbon::now()->timestamp . '_' . $request->file('image')->getClientOriginalName();
+        $request->image->move(public_path('images'), $imageName);
+
+        // Insert data on table posts
+        $urlImage = 'images/' . $imageName;
+        $this->postRepo->create([
+            'user_id' => 1,
```

Comment:
user_id này luôn là biến tĩnh hả e, hay chỉ là tạm thời thôi

Reply:
Dạ user_id này tạm thời thôi ạ.

Reply:
e cmt to do hoặc set 1 biến tạm gì đó, vd const TMP_DATA = 1 để sau này khi tìm lại sẽ tránh việc bỏ sót, để như vậy sau này rất dễ miss dẫn đến bug.

Reply:
Dạ vâng ạ.